File: 406_Queue_Reconstruction_by_Height/406_1.scala
import collection.JavaConverters._

object Solution {
    def reconstructQueue(people: Array[Array[Int]]): Array[Array[Int]] = {
        val pp = people.sortWith((a, b) => a(0) > b(0) || (a(0) == b(0) && a(1) < b(1)))
        val res: java.util.LinkedList[Array[Int]] = new java.util.LinkedList[Array[Int]]
        pp.foreach(p => res.add(p(1), p))
        res.toArray(new Array(res.size()))
        // res.asScala.toArray
    }
}

File: 406_Queue_Reconstruction_by_Height/406_1.java
class Solution {
    public int[][] reconstructQueue(int[][] people) {
        Arrays.sort(people, (o1, o2) -> o1[0] == o2[0] ? o1[1] - o2[1] : o2[0] - o1[0]);
        List<int[]> res = new LinkedList<>();
        for (int[] i : people) {
            res.add(i[1], i);
        }
        return res.toArray(new int[res.size()][2]);
    }
}
